packages/react-magma-dom/src/components/NativeSelect/NativeSelect.tsx

```diff
@@ -100,40 +132,72 @@ export const NativeSelect = React.forwardRef<HTMLDivElement, NativeSelectProps>(
+    function AdditionalContentWrapper(props) {
+      if (
+        labelPosition === LabelPosition.left ||
+        (labelPosition === LabelPosition.top && !hasLabel)
+      ) {
+        return (
+          <StyledAdditionalContentWrapper theme={theme}>
+            {props.children}
+          </StyledAdditionalContentWrapper>
+        );
+      }
+      return props.children;
+    }
+
+    const hasLabel = !!labelText;
+
     return (
-      <FormFieldContainer
-        containerStyle={containerStyle}
-        errorMessage={errorMessage}
-        fieldId={id}
-        labelPosition={labelPosition}
-        labelStyle={labelStyle}
-        labelText={labelText}
-        labelWidth={labelWidth}
-        isInverse={isInverse}
-        helperMessage={helperMessage}
-        messageStyle={messageStyle}
-        ref={ref}
-      >
-        <StyledNativeSelectWrapper
-          disabled={disabled}
-          hasError={!!errorMessage}
+      <AdditionalContentWrapper labelPosition={labelPosition}>
+        <StyledFormFieldContainer
+          additionalContent={additionalContent}
+          containerStyle={containerStyle}
+          errorMessage={errorMessage}
+          fieldId={id}
+          hasLabel={hasLabel}
+          labelPosition={labelPosition}
+          labelStyle={labelStyle}
+          labelText={
+            labelPosition !== 'left' && additionalContent ? (
+              <>
+                {labelText}
+                {labelText && additionalContent}
+              </>
+            ) : (
+              labelText
+            )
+          }
+          labelWidth={labelWidth}
           isInverse={isInverse}
-          theme={theme}
+          helperMessage={helperMessage}
+          messageStyle={messageStyle}
+          ref={ref}
         >
-          <StyledNativeSelect
-            data-testid={testId}
-            hasError={!!errorMessage}
+          <StyledNativeSelectWrapper
             disabled={disabled}
-            id={id}
+            hasError={!!errorMessage}
             isInverse={isInverse}
             theme={theme}
-            {...other}
           >
-            {children}
-          </StyledNativeSelect>
-          <DefaultDropdownIndicator disabled={disabled} />
-        </StyledNativeSelectWrapper>
-      </FormFieldContainer>
+            <StyledNativeSelect
+              data-testid={testId}
+              hasError={!!errorMessage}
+              disabled={disabled}
+              id={id}
+              isInverse={isInverse}
+              theme={theme}
+              {...other}
+            >
+              {children}
+            </StyledNativeSelect>
+            <DefaultDropdownIndicator disabled={disabled} />
+          </StyledNativeSelectWrapper>
+        </StyledFormFieldContainer>
+        {(labelPosition === 'left' && additionalContent) ||
+          (!labelText && additionalContent)}
```

Comment:
Would something like this work?

```suggestion
{additionalContent && (labelPosition === LabelPosition.left || !labelText)}
```

Reply:
I changed this out for a function, that code is old like mold

Reply:
Meh, doesn't need a function then.